### PR TITLE
ENCD-5918-filter-cart-files-to-released-analyses

### PR DESCRIPTION
**Pull request by forresttanaka**
I also fixed a couple of unrelated issues that caused Chrome debugger warnings.

### CODE REVIEW

**Comment by forresttanaka** [comment]
Preview demo cart at:
https://encd-5918-cart-analyses-pv1-fytanaka.demo.encodedcc.org/carts/a5307f66-4094-4db2-9747-dbf51182c34a/

File: src/encoded/static/components/cart/batch_download.js [hunk 1]
```diff
@@ -65,11 +65,12 @@ CartBatchDownloadButton.propTypes = {
     /** True if button disabled */
     disabled: PropTypes.bool,
     /** Called when the user clicks the button; provided by `BatchDownloadActuator` */
-    onClick: PropTypes.func.isRequired,
+    onClick: PropTypes.func,
 };
 
 CartBatchDownloadButton.defaultProps = {
     disabled: false,
+    onClick: null, // Actually required; provided by BatchDownloadActuator
```
Comment (zoldello) [inline src/encoded/static/components/cart/batch_download.js:73]: the comment is confusing. The code tell me it's not required but the comment says it is just provided by someone else...

File: src/encoded/static/components/objectutils.js [hunk 2]
```diff
@@ -948,6 +948,25 @@ export const filterForDatasetFiles = (fileList, datasets) => {
...
+/**
+ * Filter an array of files to ones included in the released analyses in `compiledAnalyses`. If
+ * none of the given compiled analyses have the "released" status, filter to the unreleased
+ * analyses files.
+ * @param {array} fileList Files to filter
+ * @param {array} compiledAnalyses Compiled analysis objects
+ * @returns {array} Filtered `fileList`
+ */
+export const filterForReleasedAnalyses = (fileList, compiledAnalyses) => {
+    if (fileList.length > 0 && compiledAnalyses.length > 0) {
```
Comment (zoldello) [inline src/encoded/static/components/objectutils.js:960]: fileList and compiledAnalyses are external to the file and they are in objectutils- Utility class everyone can call. I recommend you do defensive programming and do a falsey check like check for null; so protect the code from bad arguments.

File: src/encoded/static/components/objectutils.js [hunk 3]
```diff
@@ -948,6 +948,25 @@ export const filterForDatasetFiles = (fileList, datasets) => {
...
 
+/**
+ * Filter an array of files to ones included in the released analyses in `compiledAnalyses`. If
+ * none of the given compiled analyses have the "released" status, filter to the unreleased
+ * analyses files.
+ * @param {array} fileList Files to filter
+ * @param {array} compiledAnalyses Compiled analysis objects
+ * @returns {array} Filtered `fileList`
+ */
+export const filterForReleasedAnalyses = (fileList, compiledAnalyses) => {
```
Comment (zoldello) [inline src/encoded/static/components/objectutils.js:959]: Is this suitable in objectutil as it is not meant for general use...?

**Review by zoldello — APPROVED** [review]
Code looks okay. My comments are optional.

**Pull request merged**